janusgraph-server/src/main/java/org/janusgraph/graphdb/tinkerpop/gremlin/server/auth/JanusGraphSimpleAuthenticator.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.tinkerpop.gremlin.server.auth;
+
+import java.net.InetAddress;
+import java.util.Map;
+
+import org.apache.tinkerpop.gremlin.groovy.jsr223.dsl.credential.CredentialGraph;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticatedUser;
+import org.apache.tinkerpop.gremlin.server.auth.AuthenticationException;
+import org.apache.tinkerpop.gremlin.server.auth.Authenticator;
+import org.apache.tinkerpop.gremlin.server.auth.SimpleAuthenticator;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+
+/**
+ * A simple implementation of an {@link Authenticator} that uses a {@link Graph} instance as a credential store.
+ * Management of the credential store can be handled through the {@link CredentialGraph} DSL.
+ */
+public class JanusGraphSimpleAuthenticator extends JanusGraphAbstractAuthenticator {
+
+    private SimpleAuthenticator simpleAuthenticator;
+
+    @Override
+    public void setup(final Map<String, Object> config) {
+        super.setup(config);
+        this.backingGraph.close();
+        simpleAuthenticator = createSimpleAuthenticator();
+        simpleAuthenticator.setup(config);
+    }
+
+    @Override
+    public SaslNegotiator newSaslNegotiator(final InetAddress remoteAddress) {
+        return simpleAuthenticator.newSaslNegotiator(remoteAddress);
+    }
+
+    @Override
+    public SaslNegotiator newSaslNegotiator() {
+        return simpleAuthenticator.newSaslNegotiator();
+    }
+
+    public AuthenticatedUser authenticate(final Map<String, String> credentials) throws AuthenticationException {
+        return simpleAuthenticator.authenticate(credentials);
+    }
+
+    /**
+     * Created for testing
+     * @return
+     */
+    public SimpleAuthenticator createSimpleAuthenticator() {
```

Comment:
@sjudeng In this case, it's used in the unit tests for creating a partial mock. We don't actually want to have to create an instance of the SimpleAuthenticator in the unit tests since it's outside the scope of the unit being tested. I've seen this type of pattern before, though I'm forgetting now if it's in TinkerPop or JanusGraph.

Reply:
I see now, thanks. Can you change to `protected` and also remove (or rephrase) the testing comment since it's not just used for testing?